File: sudoku-common/common-web/src/main/java/com/zumin/sudoku/common/web/config/CommonWebProperties.java
package com.zumin.sudoku.common.web.config;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@Data
@Component
@ConfigurationProperties(prefix = "common.web")
public class CommonWebProperties {

  /**
   * 是否开启全局数据绑定
   */
  private Boolean dataBinding = true;

  /**
   * 是否开启全局异常处理
   */
  private Boolean exceptionHandler = true;

  /**
   * 是否开启全局响应处理
   */
  private Boolean response = true;

}

File: sudoku-common/common-web/src/main/java/com/zumin/sudoku/common/web/config/CommonWebProperties.kt
package com.zumin.sudoku.common.web.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConstructorBinding
@ConfigurationProperties("common.web")
data class CommonWebProperties(
  // 是否开启全局数据绑定
  val dataBinding: Boolean = true,
  // 是否开启全局异常处理
  val exceptionHandler: Boolean = true,
  // 是否开启全局响应处理
  val response: Boolean = true,
  // Swagger配置信息
  val api: ApiProperties,
) {

  data class ApiProperties(
    val title: String = "",
    val description: String = "",
  )
}